### DISCUSSION TITLE
How do I isolate the display content of different clients

### DISCUSSION THREAD

**Discussion by ponponon** [zauberzeug/nicegui#2086] [Q&A]
### Question

I am using https://github.com/zauberzeug/nicegui/blob/main/examples/ffmpeg_extract_images/main.py as an example

<img width="640" alt="图片" src="https://github.com/zauberzeug/nicegui/assets/38725104/511b0311-084e-45f1-8df3-7ce0ed808551">

But I found a problem, when I uploaded a video on my PC and took a snapshot of it; when I opened the same URL on my mobile phone, the mobile phone showed the same screen as the PC (but I didn't upload the video on my mobile phone).
 (but I didn't upload the video on my mobile). This is equivalent to different clients sharing the same data.

<img width="480" alt="图片" src="https://github.com/zauberzeug/nicegui/assets/38725104/75c58ff0-47f0-4e49-aedd-6fefd659c93d">

But if I want to have multiple clients displaying separate and isolated content, how do I achieve this? For example, I want each websocket connection to display different content, or I want each browser to display different content, or I want different ip to display different content;

**Comment by natankeddem** [comment]
Right now that example just dumps the images into a directory and displays the full content of the directory. You will need to either create a separate directory for each client/user/other identifier and display only images in those directories or keep a database which tracks what images each user is associated with.

**Comment by falkoschindler** [comment]
It's also worth mentioning that the "FFMPEG" example doesn't use `@ui.page` decorators, so every browser sees the very same "auto-index page" as NiceGUI calls it. See https://nicegui.io/documentation/section_pages_routing#auto-index_page for more information.
